addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -821,6 +821,20 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
         // We need to reload the existing type list.
         this.rerender = true;
     },
+    /**
+     * Prevents id duplication
+     * 
+     * @override
+     */
+    onClone() {
+        const originalInputEl = this.$target[0].querySelector('[id]:not(div)');
+        const cloneContainerEl = document.querySelectorAll(`#${CSS.escape(originalInputEl.id)}`)[1].closest('.s_website_form_field');
+        for (const el of cloneContainerEl.querySelectorAll('[id]:not(div)')) {
+            const newId = Math.random().toString(36).substring(2, 15); //Big unique ID
```

Comment:
I would add a new private method on the `FormEditor` to generate new ids, so that the code is not duplicated throughout the file, something like `_generateUniqueID`

Also, I think replacing all the occurences of the old field id with a new generated one would remove the onClone method dependance to the field html structure. What do you think ?
```
onClone: function () {
        const oldId = this.$target.find("[id]")[0].id;
        const newId = this._generateUniqueID();
        this.$target.html(this.$target.html().replaceAll(oldId, newId));
}
```

Reply:
I like the idea of replacing all id occurrences with new ones and creating a function that generates the id to avoid code duplication. Thanks for these tips! :smile: 
I think using the built-in replaceAll function works but for compatibility reasons I think it's better not to go for this solution (see https://caniuse.com/?search=replaceAll) 86% coverage seems too low to me.
Instead I use split() and join() which have >96% compatibility.

Reply:
I also see now that the html() function has some security risks. I don't know if in this case it is better not to use it...  :thinking:

Reply:
Don't check the % rate in caniuse, but the browser compatibility.

It is not a problem that replaceAll doesn't work on samsung internet or QQ browser...
The most important is that is is compatible with all "main" browser like edge, FF, Chrome, Safari

While animation / public code need to be the most compatible, we don't support the edition on all the support.

//cc @qsm-odoo please confirm my PoV because I already merge a replaceAll in options.js in the past. So if you are not agree with this, I need to update it too

Reply:
Security risks doesn't mean "never use", just triple check when you use it that you know the input, and that nobody can insert xss, that is is escaped and sanitized if needed  correctly

Reply:
I think that using the html() function here is not a risk in itself. At least, not more than the Label Name field which can crash if you type something like that.
`</span><script>alert("Hello")</script><span>`

Since only the site owner can edit it, I think it's not a risk in itself to use it in this case.
But I could be wrong of course 

Now if you prefer that I check what will be passed as an argument to the html() function, I can do that but then I don't understand why it's not done for other elements like the label name...

Reply:
No, It doesn't seems risky. As you said, it is only for the editor that can already add whatever as JS in the editor.
So yes, you should/could use it...

But the alert is their globally to avoid that you do an .html(partner.name) where name can be edited by the portal user (visitor) and so inject xss. And exactly to force you to think like you did here https://github.com/odoo/odoo/pull/69019#discussion_r627208490... So it works as expected, you did the job and now you  can ignore the security cli ;)

Reply:
Ok with all that is said here:
- `replaceAll` is fine to use on the editor side: we support only the last version of major browsers (no IE) to use Odoo. For code for visitors of Odoo websites... this is always blurry what we want. In that potential case, as `replaceAll` is not supported by ~10% of users *using Chrome* (most users), I would simply find an alternative since it is really easy: use `replace` with a regex, that does `replaceAll` job (I actually never used `replaceAll` ever).
- `.html(` is safe "security" safe here IMO but you should not use it: doing that, you destroy all the DOM nodes barbaric-style and re-create them. This could break other code which relies on those nodes. In the end: you need to replace IDs? Just replace IDs, don't recreate a whole portion of the DOM :wink: The editor history could also be happier to see "Clone the nodes, then replace the IDs" rather than "Clone the nodes, then reclone the nodes with other IDs" :wink: